### ISSUE TITLE
[API]: "Private" (imported) Python APIs used in across other RAPIDS projects

### ISSUE THREAD

**Issue by mroeschke** [rapidsai/cudf#15240]
This is the result of a quick audit and does not include private APIs on public objects e.g. `cudf.Series._column`

TLDR observations:

1. There seems to be a common desire to create a `Column` object (this might be solved with `pylibcudf`)?
2. There seems to be a secondary desire to create a `Buffer` object

**cuspatial**

`cudf.core.copy_types.BooleanMask` (for typing)
`cudf.core.copy_types.GatherMap` (for typing)
`cudf.core.column.ColumnBase`
`cudf.core.column.ListColumn`
`cudf.core.column.as_column`
`cudf.core.column.column_empty`
`cudf.core.buffer.acquire_spill_lock`

**cugraph**

`cudf.core.buffer.as_buffer`: Fixed by https://github.com/rapidsai/cugraph/pull/5036
`cudf.core.column.as_column`: Fixed by https://github.com/rapidsai/cudf/pull/18425
`cudf.core.column.ListColumn`: Fixed by https://github.com/rapidsai/cudf/pull/18425

**Morpheus**

`cudf.core.frame.Frame`: Fixed by https://github.com/nv-morpheus/Morpheus/pull/2211
`cudf.core.index._index_from_data`: Fixed by https://github.com/nv-morpheus/Morpheus/pull/2211
`cudf.core.subword_tokenizer.SubwordTokenizer`

**Crossfit**

`cudf.core.column.as_column`: Fixed by https://github.com/rapidsai/cudf/pull/18425
`cudf.core.index.ensure_index`: Fixed by https://github.com/rapidsai/cudf/pull/18425
`cudf.core.column.ListColumn`: Fixed by https://github.com/rapidsai/cudf/pull/18425

**Merlin**

`cudf.tests.utils.assert_eq`

**Comment by bdice** [comment]
Almost everything from `_lib` (which also means "almost everything using cudf Cython via `cimport`") will have a mirror image in `pylibcudf`. In the short term I would focus efforts on the uses of `cudf.core`, and let `cudf._lib` be resolved at a later time as pylibcudf matures.

**Comment by mroeschke** [comment]
Organizing non-`_lib` usage by "theme":

**Column APIs**

`cudf.core.column.build_column`
`cudf.core.column.column_empty`
`cudf.core.column.as_column`
`cudf.core.column.build_categorical_column`
`cudf.core.column.ListColumn`

**Buffer APIs**

`cudf.core.buffer.as_buffer`
`cudf.core.buffer.acquire_spill_lock`
`cudf.core.buffer.Buffer`

**Other APis**

`cudf.core.frame.Frame`
`cudf.core.index._index_from_data`
`cudf.core.subword_tokenizer.SubwordTokenizer`
`cudf.utils.dtypes.min_signed_type`
`cudf.tests.utils.assert_eq`

**Comment by mroeschke** [comment]
As of this comment there are 2 open PRs that will close out the known private cuDF usages

* https://github.com/nv-morpheus/Morpheus/pull/2211
* https://github.com/rapidsai/crossfit/pull/121

I can follow up with those PRs as the maintainers find time to review those, so closing
